main/externalData/balances/index.ts

```diff
@@ -200,12 +210,22 @@ export default function (store: Store) {
     store.accountTokensUpdated(address)
   }
 
+  function handleTokenBlacklistUpdate (tokens: Token[]) {
+    const tokensToRemoveSet = new Set(tokens.map(({address, chainId}) => `${chainId}:${address.toLowerCase()}`))
```

Comment:
I think we can move this mapping logic into the controller. this is why there are 2 different event emitters -- the worker can only emit raw objects but the controller can map it into its usable form, so the controller can emit this set of strings

Reply:
then the signature of this function can change to `handleTokensBlacklistUpdate(tokens: Set<TokenId>)`

Reply:
This explains why I had problems emitting a set from the worker and revertted to sending the array again!